widget/form.go

```diff
@@ -243,6 +268,29 @@ func (f *Form) setUpValidation(widget fyne.CanvasObject, i int) {
 	}
 }
 
+func (f *Form) setValidationError(err error) {
+	if err == nil && f.validationError == nil {
+		return
+	}
+
+	if (err == nil && f.validationError != nil) || (f.validationError == nil && err != nil) ||
+		!errors.Is(err, f.validationError) {
```

Comment:
Sorry, I wasn't thinking straight when I sent the old comment. The correct way is to do it like this:
```suggestion
	if !errors.Is(err, f.validationError) {
```